### PR TITLE
Payment channel opt in

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

This PR will hide the option for turning on payment channel if no funds on the channel for current `selectedAddress` and `provider.type`

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: app/components/Views/Settings/ExperimentalSettings/index.js [hunk 1]
```diff
@@ -63,15 +61,29 @@ class ExperimentalSettings extends PureComponent {
...
+		paymentChannelHasBalance: false
+	};
+
 	static navigationOptions = ({ navigation }) =>
 		getNavigationOptionsTitle(strings('app_settings.experimental_title'), navigation);
 
+	componentDidMount = async () => {
+		const paymentChannelHasBalance = await PaymentChannelsClient.addressHasBalance(this.props.selectedAddress);
+		if (paymentChannelHasBalance) this.setState({ paymentChannelHasBalance });
+		this.setState({ paymentChannelHasBalance: true });
```
Comment (danjm) [inline app/components/Views/Settings/ExperimentalSettings/index.js:84]: I think line 84 can be deleted

File: app/core/PaymentChannelsClient.js [hunk 2]
```diff
@@ -556,6 +556,25 @@ const instance = {
 	 *	Returns the current exchange rate for SAI / ETH
 	 */
 	getExchangeRate: () => (client && client.state && client.state.exchangeRate) || 0,
+	/**
+	 *	Returns whether the address has balance in channel
+	 */
+	addressHasBalance: async address => {
+		if (client) return true;
```
Comment (danjm) [inline app/core/PaymentChannelsClient.js:563]: Is it possible for a client to have been initialized but no balance sent to the address?

Comment (estebanmino) [reply app/core/PaymentChannelsClient.js:563]: yes, just enabling the experimental setting with the channel just opened

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
**Issue 1**

I am unable to enable payment channels on an account that has a balance

steps:
- I started in a state where instapay was already enabled on the app (came from develop branch) 
- built this branch and saw instapay was still enabled with the correct balance
- I then toggled off/disabled state channels on that account 
- switched to another account that didn't have any instapay balance
- killed the app and relaunched 
- switched back to the account that DOES have an instapay balance
- however, when I attempted to enable payment channels, I didn't have the option to do so 

**Issue 2** 

I don't see the option to enable payment channels when importing via seed phrase 

- started app in fresh state 
- imported my wallet via seed phrase
- once I got to the wallet view I went to settings > experimental 
- I did not see the option to enable payment channels even though my Account 1 does indeed have instapay funds

**Review by ibrahimtaveras00 — APPROVED** [review]
Issues have been resolved on both OS's, QA Passed 👍

**Pull request merged**